### DISCUSSION TITLE
Reset SIM868 modem

### DISCUSSION THREAD

**Discussion by elAlexOmDiWo** [MaJerle/lwcell#59] [Q&A]
Good day.
Good evening.
Can anyone suggest how can I reset SIM868 when using this library?
So far, I have a permanent HardFault.

Part of FreeRTOS task -

      if( event & LVGSM_RESET ) {                     // Request to reset library and modem
        ulTaskNotifyValueClear( NULL, LVGSM_RESET ); 

        for( uint8_t i = 0; i < 10; i++ ) {  
          lwgsm_mqtt_client_api_delete( mqtt_client );
          mqtt_client = NULL;
          lwgsm_network_request_detach();
          
          controlGprsRxUart( GPRS_EVT_CMD_STOP );                 // Stop RX task
          
          PWRKEY_LOW();
          vTaskDelay( 1750 );
          PWRKEY_HI();
          vTaskDelay( 3000 );

          controlGprsRxUart( GPRS_EVT_CMD_START );                // Start RX task
          
          if( lwgsmOK == lwgsm_reset_with_delay( 1000, NULL, NULL, 1 )) {
            break;
          }      
        }
      }

**Comment by MaJerle** [comment]
Where is HF triggered?

**Reply by elAlexOmDiWo** [reply]
HF ? Gps module ? As in example - 

lwgsmr_t lwgsm_ll_init(lwgsm_ll_t* ll) {
  static uint8_t memory[0x1000];
  lwgsm_mem_region_t mem_regions[] = {
    { memory, sizeof(memory) }
  };
  if (!initialized) {
    lwgsm_mem_assignmemory(mem_regions, LWGSM_ARRAYSIZE( mem_regions ));  /* Assign memory for allocations */
  }
  if (!initialized) {
    ll->send_fn = send_data;                /* Set callback function to send data */
    ll->reset_fn = reset_device;            /* Set callback for hardware reset */
  }
  initialized = 1;
  return lwgsmOK;
}

And - 
static uint8_t reset_device( uint8_t state ) {
  if (state) {                                /* Activate reset line */
    PWRKEY_LOW();
    initialized = 0;
  } 
  else {
    PWRKEY_HI();
  }
  return 1;
}

In the event modem only switch off ( before restart as in SIM868_Series_Hardware_Design_V1.07 )
![image](https://user-images.githubusercontent.com/82702074/120618844-20bd9080-c464-11eb-98bd-b3e96f96028e.png)

**Comment by MaJerle** [comment]
What code triggers hard fault?

**Reply by elAlexOmDiWo** [reply]
after this -
lwgsm_network_request_detach();

If I remove it, the library does not start because of - 
network_counter! = 0  (lwgsm_network_api.c).
Most likely I'm wrong somewhere.
How would you implement a function of cold start library with modem ?
